### PR TITLE
Deploy by using travis pages provider 

### PR DESCRIPTION
**Pull request by lzutao**
More details in https://docs.travis-ci.com/user/deployment/pages/.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,13 +12,24 @@ env:
...
 
-after_success:
-  - ./deploy.sh
+deploy:
+  - provider: pages
+    name: "Rust API Guidelines"
+    email: "guidelines@rust-lang.org"
+    skip-cleanup: true
+    local-dir: book
+    github-token: $GH_TOKEN
```
Comment (lzutao) [inline .travis.yml]: This may failed due to incorrect token type: https://docs.travis-ci.com/user/deployment/pages/#setting-the-github-token.

**Comment by KodrAus** [comment]
Thanks @lzutao!

This looks good to me, but I haven't used the GitHub Pages deployment feature of Travis before so am not totally sure how it works.

As a start, do you mind setting the `target_branch` to something like `gh_pages_test` so we can make sure we're getting the right stuff out without potentially breaking the current docs? Then I'll merge this one in and if it all looks good we'll change it to point to `gh_pages`.

File: .travis.yml [hunk 2]
```diff
@@ -8,13 +12,24 @@ env:
...
+    keep-history: false
```
Comment (KodrAus) [inline .travis.yml]:
```suggestion
    keep-history: false
    target-branch: gh_pages_test
```

Comment (lzutao) [reply .travis.yml]: Done

**Comment by lzutao** [comment]
The only things that I'm worry is incorrect token type. In that case, the deploy stage would be failed
so the old pages would be kept.

**Comment by lzutao** [comment]
This works as expected :+1:

File: .travis.yml (hunk 1 above)
Comment (lzutao) [reply .travis.yml]: done

**Pull request merged**